### ISSUE TITLE
Mouse support resize-with-center

### ISSUE THREAD

**Issue by ZeChArtiahSaher** [koekeishiya/yabai#1930]
So in default configuration there is this option:

`yabai -m config mouse_action2 resize`

Which works well however as you are resizing you are only controlling x-y axis increase at once, it would be awesome if there was an option like when you take a corner of a window and resize it while holding alt key. (I'm assuming this is native behaviour, if not basically when you resize  a corner while holding alt you are stretching/contracting x-axis and stretching-contracting y-axis). This functionality is only really useful for floating windows but I feel like it would be a logical addition, not sure exactly how the cursor should be handled....)

**Comment by koekeishiya** [comment]
I don't get what the issue is. Just move the cursor diagonally?

**Comment by ZeChArtiahSaher** [comment]
yes precisely, like what you can do when you hold alt and drag a corner of a window. It's like that implemented in altdrag on windows and it's incredibly efficient to resize windows quickly while keeping like the rough center position fixes. Especially if I'm trying to shrink it. I think with current implementation it's not possible to actually do that.

**Comment by ZeChArtiahSaher** [comment]
I just checked altdrag behavior and it's like this; the window is split into regions:
```
+---+---+---+
| ↖ | ↑ | ↗|
+---+---+---+
| ← | C | → |
+---+---+---+
| ↙ | ↓ | ↘|
+---+---+---+

So window resizing happens with respective logic. The C denotes this special resize where if you drag in positive x offset from where you start dragging it will expand in both directions and negative x will shrink. So is for the y axis. Positive y - expand, Negative y - shrink.
```
